sdk/go/telemetry/transform.go

```diff
@@ -588,13 +597,7 @@ func (s *readOnlySpan) InstrumentationLibrary() instrumentation.Library {
 
 // Resource returns information about the entity that produced the span.
 func (s *readOnlySpan) Resource() *resource.Resource {
-	if s.resource == nil {
-		return nil
-	}
```

Comment:
this nil check seems like it got dropped, not sure if important or not, just noting in case

Reply:
Pretty sure the nil check is redundant - every span and log must have a resource in OTel. Will keep an eye out for panics in case I'm wrong.